### PR TITLE
Bottom snap carousel

### PR DESCRIPTION
**Pull request by palant-dev**
Change log:

- Created a reusable component for a Snap Carousel
- Created view for Snap Carousel
- Created a model for cloths
- Added assets for icons (should be always imported as 500x500 px for best results and fast implementation)
In the future will be easy to connect a specific type of cloth to the percentage of exposed skin thanks to the model.

https://github.com/MC-Team-17/RayD/assets/113528707/9120d329-747b-4e3d-b663-3201fc71f3f4

### CODE REVIEW

File: RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    .resizable()
+                    .aspectRatio(contentMode: .fill)
+                    .frame(width: 30)
+            }
+        }
+        .frame(maxWidth: 100, maxHeight: 100, alignment: .top)
+        .onAppear {
+            // This populate the arrays with fresh icons
+            for index in 1...4 {
+                clothsUpper.append(ExposedSkin(clothIcon: "upper\(index)"))
```
Comment (yukim17) [inline RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift:55]: Maybe we can hardcode it somewhere?

File: RayD/RayD/UI/Models/ExposedSkin.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+//  RayD
+//
+//  Created by Antonio Palomba on 02/06/23.
+//
+
+import Foundation
+
+struct ExposedSkin: Identifiable {
+    var id = UUID().uuidString
+    var clothIcon: String
```
Comment (yukim17) [inline RayD/RayD/UI/Models/ExposedSkin.swift:12]: For example hardcode it here as a static variable

File: RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+//  RayD
+//
+//  Created by Antonio Palomba on 02/06/23.
+//
+
+import SwiftUI
+
+struct SnapCarouselView: View {
+
+    @State var currentIndex: Int = 0
```
Comment (yukim17) [inline RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift:12]: We need different indexes for carousels

**Review by yukim17 — CHANGES_REQUESTED** [review]

**Review by slavny11 — APPROVED** [review]

**Review by slavny11 — APPROVED** [review]

File: RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift (hunk 3 above)
Comment (palant-dev) [reply RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift]: Done!

File: RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift (hunk 1 above)
Comment (palant-dev) [reply RayD/RayD/UI/MainView/Subviews/SnapCarouselView.swift:56]: Sure I think that we can edit this part when the guys finish the icon so we can associate also the values to the specific cloth

**Review by palant-dev — COMMENTED** [review]
Added a stored property to keep track of the exposed skin

File: RayD/RayD/UI/Models/ExposedSkin.swift (hunk 2 above)
Comment (palant-dev) [reply RayD/RayD/UI/Models/ExposedSkin.swift:13]: Sorry I made a wrong correction now it should be working, try to pull the latest changes

**Review by palant-dev — COMMENTED** [review]
Added the arrays of cloths and fixed the carousel adding the slider for the head too

**Pull request merged**